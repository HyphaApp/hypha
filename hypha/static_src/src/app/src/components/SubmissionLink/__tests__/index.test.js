import React from "react";
import { mount } from "enzyme";
import sinon from "sinon";
import SubmissionLink from "../index";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });


describe("Test submission link component", () => {
    
    const subject = mount(<SubmissionLink
        submissionID = {1}
    />);
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and elements", () => {
      expect(subject.find('.submission-link').length).toBe(1);
      expect(subject.find('a').props().href).toBe(`/apply/submissions/1/`);
      expect(subject.find('a').children().length).toBe(2)
      expect(subject.find('a').text()).toEqual("Open in new tab")
    });
  
    test('render a submission link component', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });
