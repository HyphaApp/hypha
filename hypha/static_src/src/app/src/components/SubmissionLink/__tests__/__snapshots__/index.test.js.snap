// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test submission link component render a submission link component 1`] = `
<SubmissionLink
  submissionID={1}
>
  <div
    className="submission-link"
  >
    <a
      href="/apply/submissions/1/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Open in new tab
      <svg>
        <use
          xlinkHref="#open-in-new-tab"
        />
      </svg>
    </a>
  </div>
</SubmissionLink>
`;
